doc/development/style_guide.md

```diff
@@ -0,0 +1,211 @@
+# Flame Style Guide
+
+This is a general style guide for writing code within Flame and adjacent projects. We strive to
+maintain the code clean and readable -- both for the benefit of the users who will need to study
+this code in order to understand how a particular feature works or debug something that is breaking,
+and for the benefit of the current and future maintainers.
+
+This guide extends upon the official style guides of Dart and Flutter. Those guides should be
+considered an essential part of this style guide. However, whenever some rules contradict others,
+the following order of preferences apply:
+- **Flame Style Guide** (this document)
+- **[Flutter Style Guide]**
+- **[Effective Dart]**
+
+
+## Code Formatting
+
+Most of the code formatting rules are enforced automatically via the linter. Run the following
+commands to ensure the code is conformant and to fix any easy formatting problems:
+```console
+$ flutter analyze
+$ flutter format .
+```
+
+
+## Code Structure
+
+### Imports
+
+- Avoid importing groups of APIs internally, for example, importing `lib/effects.dart` just to use
```

Comment:
I don't think we want to enforce this, I think it will just be a lot more imports which doesn't add any readability, and since the compiler has tree shaking it makes no difference if more things would be included in an import.